content/loader.js

```diff
@@ -0,0 +1,17 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+chrome.runtime.sendMessage({ action: 'load' });
```

Comment:
I would say that simple is way important than performant in this case.

Content scripts are loaded in a separate execution environment which means that it can be parsed or even run in parallel. What's more the same content script is loaded for every tab so it probably needs to be processed only once and it's converted to some super optimized form by chrome. My gut feeling is that there is a chance that dynamic injection might be even less performant then this.

We can split `web3` injection and DOM augmentation though, the latter can run asynchronously after the target document was loaded (but still would prefer this to be explicitly registered in `manifest.json`). We can do proper benchmarks when we reach 1.0:)